File: ts/step6-first-class-collection/raceGame.ts

import { Car } from './car';
import { Cars } from './cars';

export class RaceGame {
    private readonly cars: Cars;

    constructor(names: string[]) {
        const carInstances = names.map(name => new Car(name));
        this.cars = new Cars(carInstances);
    }

    public moveAll(randomValues: number[]): void {
        this.cars.moveAll(randomValues);
    }

    public findWinners(): string[] {
        return this.cars.findWinners();
    }

    public getCars(): readonly Car[] {
        return this.cars.getCars();
    }
}

File: java/step6-first-class-collection/src/main/java/badcode/RaceGame.java
package badcode;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class RaceGame {
    private Cars cars;

    public RaceGame(List<String> names) {
       this.cars = Cars.from(names);
    }

    public void run(int[] randomValues) {
        cars.moveAll(randomValues);
    }

    public Cars getCars() {
        return cars;
    }


}
